pkg/tfgen/installation_docs.go

```diff
@@ -124,3 +144,192 @@ func applyEditRules(contentBytes []byte, docFile *DocFile) ([]byte, error) {
+	}
+	var cleanupErr error
+	defer func() {
+		if err := os.RemoveAll(outDir); err != nil {
+			cleanupErr = fmt.Errorf("failed to clean up configuration-example dir: %w", err)
+		}
+	}()
+	for i, block := range codeBlocks {
+		// Write the content up to the start of the code block
+		returnContent = returnContent + contentStr[startIndex:block.start]
+		nextNewLine := strings.Index(contentStr[block.start:block.end], "\n")
+		if nextNewLine == -1 {
+			//Write the inline block
+			returnContent = returnContent + contentStr[block.start:block.end] + codeFence + "\n"
+			continue
+		}
+		fenceLanguage := contentStr[block.start : block.start+nextNewLine+1]
+		code := contentStr[block.start+nextNewLine+1 : block.end]
+		// Only convert code blocks that we have reasonable suspicion of actually being Terraform.
+		if fenceLanguage == "```terraform\n" || fenceLanguage == "```hcl\n" ||
+			(fenceLanguage == "```\n" && guessIsHCL(code)) {
+
+			//Generate the main.tf file for converting the config file, since --convert-examples doesn't support
+			// the Pulumi.yaml config file
+			err := os.WriteFile(filepath.Join(outDir, "main.tf"), []byte(code), 0600)
+			if err != nil {
+				return "", fmt.Errorf("convertViaPulumiCLI: failed to write main.tf file: %w", err)
+			}
+
+			// Prepare examples
+			// Make an example to record in the cliConverter.
+			fileName := fmt.Sprintf("configuration-installation-%d", i)
+			examples[fileName] = code
+
+			// Convert to PCL with convertViaPulumiCLI.
+			// This gives us a map of translatedExamples.
+			translatedExamples, err := g.cliConverter().convertViaPulumiCLI(examples, []tfbridge.ProviderInfo{
+				g.cliConverter().info,
+			})
+			if err != nil {
+				return "", err
+			}
+
+			// Write the result to the pcls map of our cli converter.
+			g.cliConverter().pcls[code] = translatedExample{
+				PCL: translatedExamples[fileName].PCL,
+			}
+
+			exPath := examplePath{fullPath: fileName}
+			conversionResult := g.coverageTracker.getOrCreateExample(
+				exPath.String(), code)
+
+			langs := genLanguageToSlice(g.language)
+			chooserStart := `{{< chooser language "typescript,python,go,csharp,java,yaml" >}}` + "\n"
+			returnContent = returnContent + chooserStart
+			chooserEnd := "{{< /chooser >}}\n"
+
+			// Generate each language in turn, so that we can mark up the output with the correct Hugo shortcodes.
+			// TODO: we want to use pulumi code choosers in the future, but resourcedocsgen does not support this yet.
+			for _, lang := range langs {
+				langSlice := []string{lang}
+				choosableStart := fmt.Sprintf("{{%% choosable language %s %%}}\n", lang)
+				choosableEnd := "\n{{% /choosable %}}\n"
+
+				// Generate the Pulumi.yaml config file for each language
+				configFile, err := writeConfigYamlViaCLI(outDir, lang, mappingsArgs)
+				if err != nil {
+					return "", err
+				}
+				// Generate example itself
+				convertedLang, err := g.convertHCL(conversionResult, code, exPath.String(), langSlice)
```

Comment:
This is easier to read if we just inline `langSlice`.
```suggestion
				convertedLang, err := g.convertHCL(conversionResult, code, exPath.String(), []string{lang})
```